docs/proposals/PROPOSAL_E.md

```diff
@@ -0,0 +1,244 @@
+# Proposal E
+
+Cherry Pick
+
+## Description
+
+This proposal combines ideas from proposals A, B, and D to create a solution that provides an ability to use it on existing registries, while giving a value add as registries upgrade in the future.
+
+It consists of the following changes:
+
+- It adds a new media type for future use cases.
+- It defines a "reference type" field on both the new media type and existing image manifests.
+- It defines annotations for filtering artifacts.
+- It defines both an API and a digest tag syntax for querying references.
+
+The result is the following upgrade path:
+
+- Registries without new API or media type support can use the image manifest plus digest tags to create artifacts that reference other manifests.
+- As registries are upgraded, the new API can be used for more efficient queries and less tag clutter.
+  The extended Image manifest would still be used for portability to older registries.
+- When enough registries support the new media types, a transition can happen to those media types for artifacts.
+
+## Links
+
+| Description      | Link                        |
+| ---------------- | --------------------------- |
+| Proposal A       | [View](./PROPOSAL_A.md)     |
+| Proposal B       | [View](./PROPOSAL_B.md)     |
+| Proposal D       | [View](./PROPOSAL_D.md)     |
+
+## Modifications
+
+### Annotations
+
+These annotations would be added for artifacts:
+
+- `org.opencontainers.artifact.type`: type of artifact (sig, sbom, etc)
+- `org.opencontainers.artifact.description`: human readable description for the artifact
+- `org.opencontainers.artifact.created`: creation time for a manifest
+
+Additional annotations should be considered for filtering various artifact types, e.g. signature public key hash, attestation type, and sbom schema.
+
+These annotations should be considered from the current config schema:
+
+- `org.opencontainers.platform.architecture`: CPU architecture for binaries
+- `org.opencontainers.platform.os`: operating system for binaries
+- `org.opencontainers.platform.variant`: variant of the CPU architecture for binaries
+
+Existing `org.opencontainers.image.*` annotations should be reviewed to consider more generic names (e.g. replacing `image` with `manifest` or `artifact`).
+
+### JSON Schema
+
+Extend the Image Manifest with a reference field (existing registries should ignore this per OCI's extensibility requirement):
+
+```jsonc
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "config": { ... },
+  "layers": [ ... ],
+  "reference": { // new field
+    "mediaType": "application/vnd.oci.image.manifest.v1+json", // any manifest media type
+    "size": 1234,
+    "digest": "sha256:a1a1a1...",
+    "annotations": [
+      // pull up annotations from referenced manifest here
```

Comment:
I missed this during the meeting - this is something the client has to do when moving artifacts from new registries to old registries, correct?

Reply:
This one we decided to remove in the meeting. We shouldn't need it and the only annotations getting pulled up will be in the references API output by the registry.